src/persist-client/src/lib.rs

```diff
@@ -166,16 +166,24 @@ mod tests {
         Client::new(NO_TIMEOUT, location, None).await
     }
 
-    fn all_ok<'a, K, V, T, D, I>(iter: I) -> Vec<((Result<K, String>, Result<V, String>), T, D)>
+    fn all_ok<'a, K, V, T, D, I>(
+        iter: I,
+        as_of: T,
```

Comment:
agreed that we'll need`as_of` support here eventually so definitely seems fine to add, but you've got me curious what prompted adding it now?